src/main/java/com/netflix/imflibrary/writerTools/IMPAssembler.java

```diff
@@ -165,12 +164,16 @@ public AssembledIMPResult assembleIMFFromFiles(SimpleTimeline simpleTimeline, Fi
                         )
                 );
             }
+
+            // Validate and adjust the durations of the resources in the track
+            List<IMFTrackFileResourceType> adjustedResources = SequenceDurationValidator.validateAndAdjustResourceDurations(resources, simpleTimeline.getEditRate().getNumerator(), simpleTimeline.getEditRate().getDenominator());
+
```

Comment:
It might be worth considering performing the duration validation on "all tracks" together, rather than on "all Resources within one track". So maybe _after_ the `virtualTracks` list has been populated, but before the `buildIMP` call.

One advantage of this, is that it would catch _other_ kinds of track duration mismatches also. Not just fractional edit units due to EditRate mismatches, but it would also be able to catch a lot of user-error, from the provided timeline. 
As far as I can tell, `SimpleTimeline` doesn't have any sort of track duration validation. So it would be possible for the caller to (mistakenly) provide a video track with duration 30 seconds, and an audio track with duration 35 seconds.

This isn't a high priority, and is outside the scope of the problem you are _currently_ try to address. But if you end up needing to shuffle any of this logic around anyway, then it might make sense to just handle both cases in an all-purpose track-duration-validator.